iroh/src/endpoint.rs

```diff
@@ -883,38 +879,45 @@ impl Endpoint {
     //
     // Partially they return things passed into the builder.
 
-    /// Returns a stream that reports connection type changes for the remote node.
+    /// Returns a [`Watcher`] that reports the current connection type and any changes for
+    /// given remote node.
     ///
-    /// This returns a stream of [`ConnectionType`] items, each time the underlying
-    /// connection to a remote node changes it yields an item.  These connection changes are
-    /// when the connection switches between using the Relay server and a direct connection.
-    ///
-    /// If there is currently a connection with the remote node the first item in the stream
-    /// will yield immediately returning the current connection type.
+    /// This watcher allows observing a stream of [`ConnectionType`] items by calling
+    /// [`Watcher::stream()`]. If the underlying connection to a remote node changes, it will
+    /// yield a new item.  These connection changes are when the connection switches between
+    /// using the Relay server and a direct connection.
     ///
     /// Note that this does not guarantee each connection change is yielded in the stream.
-    /// If the connection type changes several times before this stream is polled only the
+    /// If the connection type changes several times before this stream is polled, only the
     /// last recorded state is returned.  This can be observed e.g. right at the start of a
     /// connection when the switch from a relayed to a direct connection can be so fast that
     /// the relayed state is never exposed.
     ///
+    /// If there is currently a connection with the remote node, then using [`Watcher::get`]
+    /// will immediately return either [`ConnectionType::Relay`], [`ConnectionType::Direct`]
+    /// or [`ConnectionType::Mixed`].
+    ///
+    /// It is possible for the connection type to be [`ConnectionType::None`] if you've
+    /// recently connected to this node id but previous methods of reaching the node have
+    /// become inaccessible.
+    ///
     /// # Errors
     ///
     /// Will error if we do not have any address information for the given `node_id`.
-    pub fn conn_type_stream(&self, node_id: NodeId) -> Result<ConnectionTypeStream> {
-        self.msock.conn_type_stream(node_id)
+    pub fn conn_type(&self, node_id: NodeId) -> Result<Watcher<ConnectionType>> {
```

Comment:
Probably out of scope for this PR.  But what would folks think of changing `fn conn_type(node_id)` into `fn network_path(node_id)`?  And Also rename `ConnectionType` into `NetworkPath`.

This name has always bothered me a bit, I'm not sure why.  Maybe because this doesn't actually have much to do with a `Connection`, it is on a different level as shown by the fact this is an `Endpoint` API.  But this is subtle and not very clear.

Reply:
I like the idea :)
I have a bunch of follow-up PR ideas brewing anyways, maybe I'll just add them to the follow-up.